src/fund.rs

```diff
@@ -0,0 +1,100 @@
+    /// Adjust period.
+    const PERIOD: u32 = 20;
+    // Factor
+    const FACTOR: u32 = 2;
+    /// Total Amount is 21000000000000000000000000 wei.
+    const TOTAL_AMOUNT: &'static str = "115EEC47F6CF7E35000000";
+    /// Initial unlock bucket is 525000000000000000000000 wei.
+    const INITIAL_BUCKET: &'static str = "6F2C4E995EC98E200000";
+
+    // There is a total fixed supply of 21 million OETHs.
+    // The blockchain unlocks 525,000 OETHs every month in the first 20 months and
+    // the monthly release is cut to 1/2 every 20 months.
+
+    /// Primary unlock token method
+    pub fn try_unlock(timestamp: u64, state: &mut State<NullBackend>) {
+        let beneficiary = Address::from_str(FundManager::BENEFICIARY).unwrap();
+        let total_amount = U256::from_str(FundManager::TOTAL_AMOUNT).unwrap();
+        let initial_bucket = U256::from_str(FundManager::INITIAL_BUCKET).unwrap();
+
+        // +---------------+----------------+-----------+-----------+
+        // | Storage field |       0:23     |   24:27   |   28:32   |
+        // +---------------+----------------+-----------+-----------+
+        // |             0 |     reserved   |    init timestamp     |
+        // +---------------+----------+-----------------+-----------+
+        // |             1 |            reserved        |   round   |
+        // +---------------+----------+-----------------+-----------+
+        // |             2 |              unlock amount             |
+        // +---------------+----------+-----------------+-----------+
+        let value = state.storage_at(&beneficiary, &H256::from(0)).unwrap();
+        let init_timestamp = value.get(24..).unwrap().read_i64::<BigEndian>().unwrap();
+        let value = state.storage_at(&beneficiary, &H256::from(1)).unwrap();
+        let mut unlock_round = value.get(28..).unwrap().read_u32::<BigEndian>().unwrap();
+        let value = state.storage_at(&beneficiary, &H256::from(2)).unwrap();
+        let mut unlock_amount: U256 = value.into();
+
+        if unlock_amount < total_amount {
+            // use last_ym keep last unlock token in which (year, month)
+            let mut last_ym = timestamp_to_mt(init_timestamp as i64);
+            last_ym.add_months(unlock_round);
+            let cur_ym = timestamp_to_mt(timestamp as i64);
+            let mut funding = U256::from(0);
+            let mut exponent = 0;
+            let mut bucket = initial_bucket;
+
+            // if last handled (year, month) still before current (year, month) and
+            // still have funding not yet unlocked.
+            while last_ym < cur_ym && unlock_amount < total_amount {
+                last_ym.add_months(1);
+                unlock_round += 1;
+                if exponent != unlock_round / FundManager::PERIOD {
+                    exponent = unlock_round / FundManager::PERIOD;
+                    bucket = initial_bucket / U256::from(FundManager::FACTOR.pow(exponent));
```

Comment:
`FACTOR.pow(exponent)` will overflow if `exponent > 31`

Reply:
Yes, you are right.
Do we need consider this scenario?
This case means (20 months)*31 / 12 months almost equal 52 years.

Reply:
We should avoid any overflow issue even if it may happen in the future.

Reply:
done. I use type U256 handle it finally.